### PR TITLE
ref(js): useLegacyStore for withGlobalSelection

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

**Review by taylangocmen — APPROVED** [review]

File: static/app/utils/withGlobalSelection.tsx [hunk 1]
```diff
@@ -9,48 +10,28 @@ type InjectedGlobalSelectionProps = {
...
-  }
+  type Props = Omit<P, keyof InjectedGlobalSelectionProps> & InjectedGlobalSelectionProps;
+
+  const WithGlobalSelection: React.FC<Props> = props => {
+    const {selection, isReady} = useLegacyStore(GlobalSelectionStore);
+
+    const selectionProps = {
+      selection,
+      isGlobalSelectionReady: isReady,
+    };
```
Comment (priscilawebdev) [inline static/app/utils/withGlobalSelection.tsx:28]:
```suggestion
```

File: static/app/utils/withGlobalSelection.tsx [hunk 2]
```diff
@@ -9,48 +10,28 @@ type InjectedGlobalSelectionProps = {
...
+
+    return <WrappedComponent {...selectionProps} {...(props as P)} />;
```
Comment (priscilawebdev) [inline static/app/utils/withGlobalSelection.tsx:30]:
```suggestion
    return <WrappedComponent selection={selection}  isGlobalSelectionReady={isReady} {...(props as P)} />;
```

**Review by priscilawebdev — APPROVED** [review]
left 1 suggestion but it looks good to me

File: static/app/utils/withGlobalSelection.tsx (hunk 2 above)
Comment (evanpurkhiser) [reply static/app/utils/withGlobalSelection.tsx:30]: I kinda like that prettier doesn't wrap it right now haha 😆

**Review by davidenwang — APPROVED** [review]
lgtm, nice stuff with `enforceActOnUseLegacyStoreHook`

**Pull request merged**